presto-hive/src/main/java/com/facebook/presto/hive/RowIDCoercer.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.common.Subfield;
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.predicate.TupleDomainFilter;
+import com.facebook.presto.common.type.BigintType;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.VarbinaryType;
+import io.airlift.slice.Slices;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+class RowIDCoercer
+        implements HiveCoercer
+{
+    private final byte[] rowIDPartitionComponent;
+    private final byte[] rowGroupID; // file name
+
+    RowIDCoercer(byte[] rowIDPartitionComponent, String rowGroupID)
+    {
+        // TODO should I copy this to avoid mutable internal state?
```

Comment:
this might be too performance sensitive an area for extra copies. @sdruzkin is the expert here. (But either way, decide and then remove the todo)

Reply:
done